go/nats/client.go

```diff
@@ -347,6 +369,8 @@ func (r *streamReader) Index(path ...uint32) (wrpc.IndexReader, error) {
 	if !ok {
 		return nil, errors.New("unknown subscription")
 	}
+	slog.Debug("Taking unsubscribe from indexedStreamReader", slog.String("path", s))
+	r.nestCloseFuncs = append(r.nestCloseFuncs, sub.Unsubscribe)
```

Comment:
Like outlined above, I think that we should use `sub.Unsubscribe` in returned  `indexedStreamReader` instead once we're done with it